pages/api/account/statistics.js

```diff
@@ -29,7 +29,21 @@ export async function getStats(username) {
 
   let profileData = {};
   try {
-    profileData = await Profile.findOne({ username });
+    const rankedProfiles = await Profile.aggregate([
+      {
+        $setWindowFields: {
+          sortBy: { views: -1 },
+          output: {
+            rank: {
+              $rank: {},
+            },
+          },
+        },
+      },
+    ]);
+    profileData = rankedProfiles.find(
```

Comment:
This will get slower as more profiles are added to the platform. Can this be done as part of the mongo aggregate query?

Reply:
> This will get slower as more profiles are added to the platform. Can this be done as part of the mongo aggregate query?

I have added the match attribute which will give the exact same result.

Reply:
Thank you 👍